### PR TITLE
Update remove api

### PR DESCRIPTION
**Pull request by JSAssassin**
Apologies, missed this when updating edvclient, doc had to be passed as an argument to delete.

Need to update bedrock web-profile, profile and profile-http once
digitalbazaar/bedrock-web-profile#13
digitalbazaar/bedrock-profile-http#28
digitalbazaar/bedrock-profile#39 are merged.

### CODE REVIEW

**Comment by aljones15** [comment]
good catch this does seem to line up with the latest interface for the `edvDocument`: 

https://github.com/digitalbazaar/edv-client/blob/4a70c37890a54885f52e5f28cd11b8a4dafab9c1/EdvDocument.js#L106-L122

**Review by aljones15 — APPROVED** [review]
I'm approving this, but would like to see tests for this in this project. That can be done in a separate issue as this is a pretty good fix. Good catch and no need to apologize you've seen how bad my code can be.

**Comment by JSAssassin** [comment]
Thank you for reviewing this @aljones15 
> p.s. the jsdoc for both delete methods in EdvDocument and EdvClient seem to be out of whack. I'm assuming this was tested?
jsdocs has been updated in this PR https://github.com/digitalbazaar/edv-client/pull/73, could you please take a look at it and merge it if its good to go? (I think the only issue there was that jsdoc-to-markdown didn't seem to be working)
> Are there any tests in `web-profile-manager` for this? I'm not seeing any.
I don't see it too, but I can add one in a separate PR :)

File: Collection.js [hunk 1]
```diff
@@ -76,7 +76,9 @@ export default class Collection {
       return false;
     }
     const edvDoc = await this._getEdvDocument({id: existing.id});
-    return edvDoc.delete();
+    const docResult = await edvDoc.read();
```
Comment (dlongley) [inline Collection.js:79]: This gets the document twice -- see the line above that gets the existing doc already. We should remove that line up there and just read the doc once -- and handle any exception properly (if it doesn't exist) and then remove the doc.

**Review by dlongley — CHANGES_REQUESTED** [review]

File: Collection.js [hunk 2]
```diff
@@ -76,7 +76,7 @@ export default class Collection {
       return false;
     }
     const edvDoc = await this._getEdvDocument({id: existing.id});
-    return edvDoc.delete();
+    return edvDoc.delete({doc: edvDoc});
```
Comment (dlongley) [inline Collection.js:79]: I don't think this is right. I think the `doc` it is expecting is not an `EdvDocument` instance, rather, the JSON document that has the sequence number on it, etc. If this is passing tests, I suspect our tests are faulty. Shouldn't it be this?

```suggestion
    return edvDoc.delete({doc: existing});
```

**Review by dlongley — CHANGES_REQUESTED** [review]

File: Collection.js (hunk 2 above)
Comment (JSAssassin) [reply Collection.js]: @dlongley I misunderstood but now it makes sense. There are no tests currently in the repo, so, I'm using wallet/issuer to test this. you're right the change I made doesn't work. I've updated it to changes suggested. Thanks.

**Review by dlongley — APPROVED** [review]

**Comment by aljones15** [comment]
@JSAssassin I think this is good to merge and then you can ping someone for a release.

**Pull request merged**